test/solc/CommandLineParser.cpp

```diff
@@ -75,12 +73,9 @@ BOOST_AUTO_TEST_CASE(no_options)
 	expectedOptions.modelChecker.initialize = true;
 	expectedOptions.modelChecker.settings = {};
 
-	stringstream serr;
-	optional<CommandLineOptions> parsedOptions = parseCommandLine(commandLine, serr);
+	CommandLineOptions parsedOptions = parseCommandLine(commandLine);
 
-	BOOST_TEST(serr.str() == "");
-	BOOST_REQUIRE(parsedOptions.has_value());
-	BOOST_TEST(parsedOptions.value() == expectedOptions);
```

Comment:
Not relevant here, but you can actually compare `optional<T>` with `T` directly, so this and the previous line would be equivalent to
`BOOST_TEST(parsedOptions == expectedOptions);`

Reply:
I think that it won't work with this macro though because boost requires both types to be printable so I'd have to define `<<` also for the optional version. Well, maybe a generic `<<` for printing `optional<T>` you had in one of your PRs would work for that.